stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -46,15 +46,37 @@ julia> vals, vecs = F; # destructuring via iteration
 julia> vals == F.values && vecs == F.vectors
 true
 ```
+
+Depending on the constructing algorithm, an object `F::Eigen` may optionally also contain:
+
+* the left eigenvectors as the columns of the matrix `F.vectorsl`,
+* the reciprocal condition numbers of the eigenvalues in the vector `F.rconde`,
+* the reciprocal condition numbers of the eigenvectors in the vector `F.rcondv`,
+* the norm (usu. `p=1`) of the balanced form of the matrix in `F.balnorm`.
 """
-struct Eigen{T,V,S<:AbstractMatrix,U<:AbstractVector} <: Factorization{T}
+struct Eigen{T,V,S<:AbstractMatrix,U<:AbstractVector,R<:AbstractVector,B} <: Factorization{T}
     values::U
     vectors::S
-    Eigen{T,V,S,U}(values::AbstractVector{V}, vectors::AbstractMatrix{T}) where {T,V,S,U} =
-        new(values, vectors)
+    vectorsl::S
+    unitary::Bool # indicates that the vectors form a unitary matrix (normality)
+    rconde::R
+    rcondv::R
+    balnorm::B
+    Eigen{T,V,S,U,R,B}(values::AbstractVector{V}, vectors::AbstractMatrix{T}, vectorsl::AbstractMatrix{T}, unitary::Bool, rconde::R, rcondv::R, abnrm::B) where {T,V,S,U,R,B} =
+        new(values, vectors, vectorsl, unitary, rconde, rcondv, abnrm)
+end
+
+# Normal matrices are exceptional, so it's safer to force their processors to do extra work.
+function Eigen(values::AbstractVector{V}, vectors::AbstractMatrix{T}, vectorsl=similar(vectors,0,0), rce=zeros(real(T),0), rcv=zeros(real(T), 0), balnorm=0/zero(real(T)); normal=false) where {T,V}
+    if normal
+        n = length(values)
+        vectorsl = vectors
+        balnorm = maximum(abs.(values))
+        rce = ones(real(T),n)
+        rcv = ones(real(T),n)
```

Comment:
Is this right? Doesn't this depend on the eigenvalue gaps?

Reply:
My assignment to `rcv` is certainly wrong, so I will change that to undefined (empty) for now. (The symmetric wrapper could be adapted to compute it but this is already a messy PR, so not today.)

My understanding is that the LAPACK symmetric algorithms only claim the `rce=1` (Bauer-Fike) accuracy except in special situations, so this seemed safe.

Reply:
The conditioning is a property of the mathematical problem, not of the algorithm, so rce=1 in the symmetric case is basically just a restatement of the Bauer-Fike bound, no? Though it's not clear to me why these fields are useful, as they can be obtained by a postprocessing of the factorization.